src/engraving/dom/dynamic.cpp

```diff
@@ -423,19 +423,33 @@ void Dynamic::manageBarlineCollisions()
 //---------------------------------------------------------
 
 void Dynamic::setDynamicType(const String& tag)
+{
+    const DynamicType dt = parseDynamicText(tag);
+
+    if (dt == DynamicType::OTHER) {
+        LOGD("setDynamicType: other <%s>", muPrintable(tag));
+    }
+
+    setDynamicType(dt);
+}
+
+DynamicType Dynamic::parseDynamicText(const String& tag)
 {
     std::string utf8Tag = tag.toStdString();
-    size_t n = DYN_LIST.size();
-    for (size_t i = 0; i < n; ++i) {
-        if (TConv::toXml(DynamicType(i)).ascii() == utf8Tag || DYN_LIST[i].text == utf8Tag) {
-            setDynamicType(DynamicType(i));
-            setXmlText(String::fromUtf8(DYN_LIST[i].text));
-            return;
+    std::regex dynamicRegex(R"((?:<sym>.*?</sym>)+|(?:\b)[fmnprsz]+(?:\b(?=[^>]|$)))");
```

Comment:
```suggestion
    const std::regex dynamicRegex(R"((?:<sym>.*?</sym>)+|(?:\b)[fmnprsz]+(?:\b(?=[^>]|$)))");
```

Same goes for `match` and `matchStr`